hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/parse/QueryParser.java

```diff
@@ -35,54 +36,44 @@
 import org.apache.calcite.sql.util.SqlVisitor;
 import org.apache.calcite.sql.validate.SqlConformance;
 
-import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 import java.util.List;
 
+import static com.hazelcast.jet.impl.util.Util.JET_IS_DISABLED_MESSAGE;
+
 /**
  * Performs syntactic and semantic validation of the query.
  */
 public class QueryParser {
 
     private final HazelcastTypeFactory typeFactory;
     private final CatalogReader catalogReader;
-    private final SqlConformance conformance;
     private final SqlConformance jetConformance;
     private final List<Object> arguments;
 
-    private final SqlBackend sqlBackend;
     private final SqlBackend jetSqlBackend;
 
     public QueryParser(
             HazelcastTypeFactory typeFactory,
             CatalogReader catalogReader,
-            SqlConformance conformance,
             SqlConformance jetConformance,
             List<Object> arguments,
-            @Nonnull SqlBackend sqlBackend,
             @Nullable SqlBackend jetSqlBackend
     ) {
         this.typeFactory = typeFactory;
         this.catalogReader = catalogReader;
-        this.conformance = conformance;
         this.jetConformance = jetConformance;
         this.arguments = arguments;
-
-        this.sqlBackend = sqlBackend;
         this.jetSqlBackend = jetSqlBackend;
     }
 
     public QueryParseResult parse(String sql) {
         try {
-            try {
-                return parse(sql, sqlBackend, conformance);
-            } catch (Exception e) {
-                // TODO: once IMDG engine is removed, move the check (and fail fast) to SqlServiceImpl?
-                if (jetSqlBackend != null) {
-                    return parse(sql, jetSqlBackend, jetConformance);
-                } else {
-                    throw e;
-                }
+            if (jetSqlBackend != null) {
+                return parse(sql, jetSqlBackend, jetConformance);
+            } else {
+                // TODO: move this check to SqlServiceImpl.
+                throw new JetDisabledException(JET_IS_DISABLED_MESSAGE);
```

Comment:
I assume my review is required because I suggested JesDisabledException should be thrown here. If so it looks good to me.